apps/src/componentLibrary/checkboxDropdown/CheckboxDropdown.tsx

```diff
@@ -0,0 +1,193 @@
+
+export interface CheckboxDropdownProps {
+  /** CheckboxDropdown name */
+  name: string;
+  /** CheckboxDropdown color */
+  color?: 'white' | 'black';
+  /** CheckboxDropdown size */
+  size: ComponentSizeXSToL;
+  /** CheckboxDropdown disabled state */
+  disabled?: boolean;
+  /** CheckboxDropdown label */
+  labelText: string;
+  /** CheckboxDropdown options */
+  allOptions: {value: string; label: string; isOptionDisabled?: boolean}[];
+  /** CheckboxDropdown checked options */
+  checkedOptions: string[];
+  /** CheckboxDropdown onChange handler */
+  onChange: (args: React.ChangeEvent<HTMLInputElement>) => void;
+  /** CheckboxDropdown onSelectAll handler */
+  onSelectAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  /** CheckboxDropdown onClearAll handler */
+  onClearAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+/**
+ * ### Production-ready Checklist:
+ * * (✔) implementation of component approved by design team;
+ * * (✔) has storybook, covered with stories and documentation;
+ * * (✔) has tests: test every prop, every state and every interaction that's js related;
+ * * (see apps/test/unit/componentLibrary/DropdownMenuTest.jsx)
+ * * (?) passes accessibility checks;
+ *
+ * ###  Status: ```Ready for dev```
+ *
+ * Design System: Checkbox Dropdown Component.
+ * Used to render checkbox (multiple choice) dropdowns.
+ */
+const CheckboxDropdown: React.FunctionComponent<CheckboxDropdownProps> = ({
+  name,
+  labelText,
+  allOptions,
+  checkedOptions = [],
+  onChange,
+  onSelectAll,
+  onClearAll,
+  disabled = false,
+  color = 'black',
+  size = 'm',
+}) => {
+  const {activeDropdownName, setActiveDropdownName} = useDropdownContext();
+  const dropdownRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const handleClickOutside = useCallback(
+    (event: Event) => {
+      if (
+        activeDropdownName &&
+        dropdownRef.current &&
+        event.target instanceof Node &&
+        !dropdownRef.current.contains(event.target)
+      ) {
+        setActiveDropdownName('');
+      }
+    },
+    [dropdownRef, setActiveDropdownName, activeDropdownName]
+  );
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleClickOutside);
+    };
+  }, [handleClickOutside]);
+
+  const toggleDropdown = useCallback(() => {
+    if (activeDropdownName !== name) {
+      setActiveDropdownName(name);
+    } else {
+      setActiveDropdownName('');
+    }
+  }, [name, activeDropdownName, setActiveDropdownName]);
+
+  const isOpen = useMemo(
+    () => activeDropdownName === name,
+    [activeDropdownName, name]
+  );
+
+  // Collapse dropdown if 'Escape' is pressed
+  const onKeyDown: (e: KeyboardEvent) => void = e => {
+    if (e.keyCode === 27) {
```

Comment:
nit: could add this key code to [constant `KeyCodes`](https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci8yNTcwZTNiNjNhMWE2ZTNiODgzZDQ5ODlkZTE3ZjFkOGNhMzkwYjIyP3VybD1naXQlNDBnaXRodWIuY29tJTNBY29kZS1kb3Qtb3JnJTJGY29kZS1kb3Qtb3JnLmdpdA%3D%3D?origin=gitlens)

Reply:
Thanks for the hint!
keyCode is now actually deprecated, so I've rewritten this to use e.key instead which is more clear (see followup pr)